### ISSUE TITLE
Upgrading to communication-react 1.7 beta version is breaking a bunch of unit tests

### ISSUE THREAD

**Issue by sayayinR** [Azure/communication-ui-library#3562]
**Describe the bug; what happened?**
After upgrading to communication-react beta version and its chat and calling dependencies, existing tests that use react testing library user event are all breaking with the following error:
```
 TypeError: Cannot read properties of null (reading 'document')
      at la (../../node_modules/@azure/communication-react/dist/dist-esm/northstar-wrapper/src/index.js:2:1579
```
      
If I revert back to the old version of communication react i.e 1.5.0 and other lower versions of communication-calling and communication-chat, all tests start to pass.
**What are the steps to reproduce the issue?**
After installing the packages mentioned above. Just write a unit test and use jest with `@testing-library/user-event` for testing some behavior around click on a button. e.g:
```
    const user = userEvent.setup();
    await user.click(screen.getByText("TestButton"));
```
**What behavior did you expect?**

**If applicable, provide screenshots:**
![image](https://github.com/Azure/communication-ui-library/assets/11239820/9697b6c9-aafc-4d5c-9576-30c47dc70a08)

**In what environment did you see the issue?**

- `@azure/communication-react` npm package version (if applicable): 1.7.0-beta.2
- `@azure/communication-calling` npm package version (if applicable): 1.15.1-beta.1
- `@azure/communication-chat` npm package version (if applicable): 1.3.2-beta.3

- OS & Device: Windows 11

**Is there any additional information?**

**Comment by JamesBurnside** [comment]
Hi @sayayinR, can you share your jest config file as well as your node version? Could you also check your jest configuration has testEnvironment set to a browser environment, such as jsdom
```js
testEnvironment: 'jsdom'
```

**Comment by sayayinR** [comment]
@JamesBurnside thanks for the reply, 
Node version is 18.17.1
Test env is set to:   "jest-environment-jsdom",

Below is the whole jest.config

```
import type { Config } from "@jest/types";

const config: Config.InitialOptions = {
  roots: ["<rootDir>/src"],
  preset: "ts-jest",
  setupFiles: ["<rootDir>/.test/jest.init.ts"],
  setupFilesAfterEnv: ["@testing-library/jest-dom/extend-expect", "jest-extended/all"],
  testEnvironment: "jest-environment-jsdom",
  testRegex: "\\.(test|spec)\\.tsx?$",
  moduleFileExtensions: ["ts", "tsx", "js", "jsx", "json"],
  collectCoverageFrom: ["./src/**", "!./src/**/*.stories.tsx", "!./src/**/*.dev.stories.tsx", "./test/**"],
  coverageReporters: ["cobertura", "clover", "json", "lcov", "text"],
  moduleNameMapper: {
    "\\.s?css$": "identity-obj-proxy",
    // Force module uuid to resolve with the CJS entry point, because Jest does not support package.json.exports. See https://github.com/uuidjs/uuid/issues/451
    uuid: require.resolve("uuid"),
    // Fix for an existing issue with proxy-memoize when "esModuleInterop": true, https://github.com/dai-shi/proxy-memoize/issues/42
    "^proxy-memoize$": "<rootDir>/node_modules/proxy-memoize/dist/wrapper.cjs",
    nanoid: require.resolve("nanoid"),
  },
  testTimeout: 60000,
};

export default config;

```

**Comment by JamesBurnside** [comment]
Thanks for sharing that. Some of our developers have hit this issue internally as well and found any function from `@testing-library/user-event` that calls `fireEvent.mouseDown` to trigger this error, we found all keyboard events from that same library to work.

Our best reasoning to what is happening is an error occurs causing jest to clean up the jsdom objects, then an async function/microtask is trying to subsequently access the package and hence the document object (the changes in communication-react v1.7 triggered document access at import time). We hoped that https://github.com/jestjs/jest/issues/12670 would resolve this issue but unfortunately it did not.

We ultimately avoided using the userevent.click events and opted for either keyboard events or `fireEvent.click` APIs instead as a short-term workaround and this is our best suggestion for what to do today. We are open to implementing a more concrete fix if you can determine the underlying reason for what is occurring.

However, we are in the process of removing the package were that error originates #3427, I've test on this branch and we no longer hit see this issue with the fluent northstar library removed. Once this PR is in and we release a new version that should resolve this issue for you. Please note this is a larger change that could take some time to check in.

**Comment by sayayinR** [comment]
Sounds good, I ll try out the the couple of options you suggested. Thank you for a prompt response.

**Comment by JamesBurnside** [comment]
As #3578 has now been completed and released in 1.9.0-beta.1 this should be fixed. We are currently working on releasing 1.9.0 with the fix also.
